io-engine/src/bdev/nvmx/qpair.rs

```diff
@@ -509,6 +525,21 @@ impl<'a> Connection<'a> {
             w.send(res.clone())
                 .expect("Failed to notify a connection waiter");
         });
+
+        // Wake up the waiter who received EAGAIN because this async connect was ongoing.
+        CHAN_ITER_MARK.with(|c| {
+            if let Some(cstored) = unsafe { &mut *c.get()}.take() {
+                let chan = unsafe {&mut *(cstored.0 as *mut NexusChannel)};
+                let iter = unsafe {&mut *(cstored.1)};
+                warn!("Consume marker and continue traversing channels: {:?} - {}", chan, Mthread::current().unwrap().name());
+                match chan.reconnect_all() {
```

Comment:
this seems quite risky to run outside of the channel traversal..
Are we sure we're connecting the right channel here?